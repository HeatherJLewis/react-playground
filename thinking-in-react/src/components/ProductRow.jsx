import React from "react";

export default function ProductRow({ item }) {
  // props.item.name
  return <div>{item.name}</div>;
}

//
// {props :
// { item: {
//   name: 'someName'
// }}
// }
