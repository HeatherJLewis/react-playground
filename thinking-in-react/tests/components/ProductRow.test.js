import ProductRow from "../../src/components/ProductRow";
import { render } from "@testing-library/react";

const fakeObject = {
  item: {
    name: "someName",
  },
};
const renderComponent = (params) => {
  render(<ProductRow {...params} />);
};

describe("ProductRow", () => {
  it("should render as expected", () => {
    const component = renderComponent(fakeObject);
    expect(component).toMatchSnapshot();
  });
});
