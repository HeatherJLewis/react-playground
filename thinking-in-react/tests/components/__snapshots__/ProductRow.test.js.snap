// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRow should render as expected 1`] = `undefined`;
